### ISSUE TITLE
OnBeforeSave result ignored

### ISSUE THREAD

**Issue by clach04** [moltenform/scite-with-python#3]
I experimented with the OnBeforeSave event, which https://www.scintilla.org/SciTEExtension.html documents as:

> OnBeforeSave is called before saving the file and an extension may implement file saving itself and return true to prevent the default file save code from executing.

I've tried returning True (and even 1, False, and 0 for completeness) but each time the file is saved to disk:

    def OnBeforeSave(filename):
            print('plugin %s file %s' % (__file__, filename))  # goes to output pane
            #return True
            #return False
            return 1
            #return 0

The event does fire as output is displayed to the output pane, i.e. the register is successful.

is this a bug in scite-with-python, doc issue, bug in code above?

**Comment by benf-write** [comment]
I would try the following and see if they work,

* Returning the special string `"StopEventPropagation"`
* Or, raising an unhandled exception.

Let me know if this helps!

**Comment by clach04** [comment]
@benf-write  yep the magic value did it! I did not try and unhandled exception.

I guess it's deliberate based on https://github.com/moltenform/scite-with-python/blob/6a685e55d159cd8850a6765fcc3e37f1df587403/src/scite/scite/src/PythonExtension.cxx#L625 ?

This should give me enough to get encryption/decryption working with Scite and my library https://github.com/clach04/puren_tonbo (once I get a few hours to duct tape them together) :-)

Thanks again for this project and for the help!

**Comment by clach04** [comment]
For completeness, working version(s):

    from scite_extend_ui import ScConst

    def OnBeforeSave(filename):
            print('plugin %s file %s' % (__file__, filename))  # goes to output pane
            return False  # Scite will handle save
            #return "StopEventPropagation"  # works
            #return ScConst.StopEventPropagation  # works, tells Scite to NOT save
